src/Rules.Framework.Rql/Pipeline/Parse/Strategies/OperatorParseStrategy.cs

```diff
@@ -0,0 +1,58 @@
+namespace Rules.Framework.Rql.Pipeline.Parse.Strategies
+{
+    using System;
+    using System.Collections.Generic;
+    using Rules.Framework.Rql.Ast.Segments;
+    using Rules.Framework.Rql.Tokens;
+
+    internal class OperatorParseStrategy : ParseStrategyBase<Segment>, ISegmentParseStrategy
+    {
+        public OperatorParseStrategy(IParseStrategyProvider parseStrategyProvider)
+            : base(parseStrategyProvider)
+        {
+        }
+
+        public override Segment Parse(ParseContext parseContext)
+        {
+            var currentToken = parseContext.GetCurrentToken();
+            var operatorTokens = new List<Token>(2)
+            {
+                currentToken,
+            };
+
+            switch (currentToken.Type)
+            {
+                case TokenType.AND:
+                case TokenType.ASSIGN:
+                case TokenType.EQUAL:
+                case TokenType.GREATER_THAN:
+                case TokenType.GREATER_THAN_OR_EQUAL:
+                case TokenType.IN:
+                case TokenType.LESS_THAN:
+                case TokenType.LESS_THAN_OR_EQUAL:
+                case TokenType.MINUS:
+                case TokenType.NOT_EQUAL:
+                case TokenType.OR:
+                case TokenType.PLUS:
+                case TokenType.SLASH:
+                case TokenType.STAR:
+                    break;
+
+                case TokenType.NOT:
+                    if (!parseContext.MoveNextIfNextToken(TokenType.IN))
+                    {
+                        parseContext.EnterPanicMode("Expected token 'in'.", parseContext.GetNextToken());
```

Comment:
`PanicMode`? 😆

Reply:
Yes, that's the concept, I'm not responsible for the name 😄 [See here](https://craftinginterpreters.com/parsing-expressions.html#panic-mode-error-recovery) if you want to know more 😉